### ISSUE TITLE
Not working properly in VS Code monorepo setup

### ISSUE THREAD

**Issue by elliason** [mrmckeb/typescript-plugin-css-modules#195]
I somehow can't get this plugin to work with VS Code properly.

I have an monorepo typescript app. One of my packages use scss modules.
The TS & plugin versions:
![image](https://user-images.githubusercontent.com/6887123/215755229-9751b301-2e63-47e1-bf09-728d064c3933.png)

The package tsconfig:
![image](https://user-images.githubusercontent.com/6887123/215752956-1f6da2dc-da28-4b71-a9df-60f70af7f71e.png)
The module .d.ts:
![image](https://user-images.githubusercontent.com/6887123/215753172-ba141adc-94c0-4015-934f-5cb3eb5b7065.png)
The scss file classname:
![image](https://user-images.githubusercontent.com/6887123/215753357-47909cdd-b65f-41e2-b614-0cea3a536860.png)

Now, if I use TS version from VS code (not the workspace version), autocomplete looks fine:
![image](https://user-images.githubusercontent.com/6887123/215753841-ec817d36-035a-47fb-a8c8-35ffde71c0c1.png)
But if I insert wrong classname there is no type hint, no error message:
![image](https://user-images.githubusercontent.com/6887123/215753765-9220bc95-1500-4276-9750-1e0be25ff447.png)
Also no type information on the "style" object (as should be according to your example.gif
![image](https://user-images.githubusercontent.com/6887123/215754123-878f1075-e554-466c-a5aa-6e8678be604f.png)

If I switch to workspace TS version I also get autocompletion working fine, but the result is marked as wrong:
![image](https://user-images.githubusercontent.com/6887123/215754677-136d9c26-84a6-4688-a029-074c14644d3d.png)

I would expect behavior similar to what I have seen in your example.gif. Correct type information on the style object and correct type hints for classnames.

I would be grateful for any idea what I might have done wrong.

**Comment by johndalvik** [comment]
Have you tried this?
- The typescript version on the project should be configured to use the workspace version, not VS codes.
- This can be set with the "Select typescript version..." command.

**Comment by elliason** [comment]
@johndalvik  Yes, as I have written as the last point above "If I switch to workspace TS version I also get autocompletion working fine, but the result is marked as wrong" (even if its the correct className). So remaining with the VS Codes version actually works better for me, although its not perfect as it shlould be according to example - auto-completion yes but no type information and validation after that.
I have also asked my colleagues with similar setups and they have the very same situation. So this might be a problem for more people.

**Comment by krutoo** [comment]
same issue

**Comment by mrmckeb** [comment]
Hi @elliason, we use this plugin internally at Vercel in our monorepo - we've used it with Yarn and pnpm.

You definitely need to use the workspace version of TypeScrip, as @johndalvik said. This is because VSCode won't load plugins if you don't.

It seems that you're still having issues in VS Code even with that enabled. Are you able to provide a small reproduction of the issue? I can then dig in and hopefully resolve it.

**Comment by avs-doom** [comment]
I get the same behavior, I noticed such a thing that if you remove the plugin from tsconfig, then everything starts to work correctly and c typescript from node_modules

**Comment by mrmckeb** [comment]
I'll need a reproduction to debug this properly. I use this in multiple monorepos and it works as expected, so I need more information to understand what conditions cause this issue.

If anyone can help provide that, I'll be more than happy to debug and create/document a fix.

**Comment by elliason** [comment]
@mrmckeb 
Hi, I have created some minimal monorepo as example. https://github.com/elliason/ts-css-modules-monorepo-example 
This is based on different setup than the initial example, does not use custom Webpack settings to bundle code, but rather depends on Vite. It seems that Vite is providing with type definitions for css modules, so there is no need to declare custom .d.ts file. But I still can not get correct classname linting in VSCode. The type information about style object properties is lost, there are only strings.
Look for component `packages/ui/src/components/atoms/button/Button.tsx` There is classname `styles.somethingThatsNotCorrect` used without problem - but in fact this classname does not exist at all.
There are however some differences to the initial example that was custom Webpack build - now it does not matter if I use VS Code TS workspace version. The result still looks the same. Previously using TS Workspace version was causing additional bugs.

**Comment by mrmckeb** [comment]
HI @elliason,

I finally had some time to look into this today. I found the bug, and luckily this was resolved as a part of our recent rewrite for TypeScript 5.0.

You don't need to upgrade to TypeScript 5, but v5 of this plugin does fix it when I upgrade that monorepo example.
<img width="513" alt="image" src="https://user-images.githubusercontent.com/5043083/230857292-8afad9f3-c194-4c38-b0eb-538f6b86752a.png">

Sorry you hit this, and thanks for taking the time to report this thoroughly.
